controllers/token.go

```diff
@@ -229,3 +229,65 @@ func (c *ApiController) TokenLogout() {
 	c.Data["json"] = wrapActionResponse(flag)
 	c.ServeJSON()
 }
+
+// IntrospectToken
+// @Title IntrospectToken
+// @Description The introspection endpoint is an OAuth 2.0 endpoint that takes a
+//  parameter representing an OAuth 2.0 token and returns a JSON document
+//  representing the meta information surrounding the
+//  token, including whether this token is currently active.
+//  This endpoint only support Basic Authorization.
+// @Param body body {object.TokenIntrospectionRequest} true "the request body"
+// @Success 200 {object} object.IntrospectionResponse The Response object
+// @router /login/oauth/introspect [post]
+func (c *ApiController) IntrospectToken() {
+	var body object.TokenIntrospectionRequest
+	err := json.Unmarshal(c.Ctx.Input.RequestBody, &body)
+	clientId, clientSecret, ok := c.Ctx.Request.BasicAuth()
+	if !ok {
+		util.LogWarning(c.Ctx, "Basic Authorization parses failed")
+		c.Data["json"] = Response{Status: "error", Msg: "Unauthorized operation"}
+		c.ServeJSON()
+		return
+	}
+	application := object.GetApplicationByClientId(clientId)
+	if application == nil || application.ClientSecret != clientSecret {
+		util.LogWarning(c.Ctx, "Basic Authorization failed")
+		c.Data["json"] = Response{Status: "error", Msg: "Unauthorized operation"}
+		c.ServeJSON()
+		return
+	}
+	token := object.GetTokenByTokenAndApplication(body.Token, application.Name)
+	if token == nil {
+		util.LogWarning(c.Ctx, "application: %s can not find token", application.Name)
+		c.Data["json"] = &object.IntrospectionResponse{Active: false}
+		c.ServeJSON()
+		return
+	}
+	jwtToken, err := object.ParseJwtTokenByApplication(body.Token, application)
+	if err != nil || jwtToken.Valid() != nil {
+		// and token revoked case. but we not implement
+		// TODO: 2022-03-03 add token revoked check, when we implemented the Token Revocation(rfc7009) Specs.
+		// refs: https://tools.ietf.org/html/rfc7009
```

Comment:
Here is to delete the token in the database based on the token? If it is, can use object.DeleteTokenByAceessToken.

Reply:
this is anathor specs. [OAuth 2.0 Token Revocation(rfc7009)](https://datatracker.ietf.org/doc/html/rfc7009).

we didn't implement it

Reply:
> this is anathor specs. [OAuth 2.0 Token Revocation(rfc7009)](https://datatracker.ietf.org/doc/html/rfc7009).
> 
> we didn't implement it

when we implement it, this place need to check the token has been revoked.